atc/configwarning.go

```diff
@@ -1,6 +1,24 @@
 package atc
 
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
 type ConfigWarning struct {
 	Type    string `json:"type"`
 	Message string `json:"message"`
 }
+
+var validIdentifiers = regexp.MustCompile(`^[\p{Ll}\p{Lt}\p{Lm}\p{Lo}][\p{Ll}\p{Lt}\p{Lm}\p{Lo}\d\-.]*$`)
+
+func ValidateIdentifier(identifier string, context ...string) *ConfigWarning {
+	if identifier != "" && !validIdentifiers.MatchString(identifier) {
+		return &ConfigWarning{
+			Type:    "invalid_identifier",
+			Message: fmt.Sprintf("%s: %s", strings.Join(context, ""), fmt.Sprintf("'%s' is not a valid identifier", identifier)),
```

Comment:
I wonder if we can provide more context as to what a valid identifier is, since to a new user (or to an old user who isn't familiar with the valid identifiers RFC), it might not be obvious. While we will have this documented, perhaps it's better to tell the user what exactly was wrong with their identifier, or give a brief summary of what a valid identifier is. That could just be the responsibility of `fly` when it encounters an `invalid_identifier` warning, or we could encode this information into the `Message` itself...not sure what the best approach is.

Since the validation logic is pretty simple, maybe it's worth not using a regex at all (which looks a bit intimidating at this point) and instead do the validation manually. This way, we can give more fine-grained warning messages that can help users out (e.g. `'some_job' is not a valid identifier: illegal character '_'` or `'123abc' is not a valid identifier: must start with a letter`)

What do you think about that approach?

Reply:
Sure! Let's add a brief summary for _must start with_ and _illegal character_. Should we print a single item (the left most one) or a list?

> That could just be the responsibility of `fly` when it encounters an `invalid_identifier` warning, or we could encode this information into the `Message` itself...not sure what the best approach is.
> 

Encoding the details in the `Message` will keep consistency between client-side and server-side validation.

> Since the validation logic is pretty simple, maybe it's worth not using a regex at all (which looks a bit intimidating at this point) and instead do the validation manually.

If it's ok with you, I'd keep using a regex for now... There's a simple way to catch the details using them.